jupyter_server/services/kernels/kernelmanager.py

```diff
@@ -243,7 +243,7 @@ async def _async_start_kernel(  # type:ignore[override]
             kernel.reason = ""  # type:ignore[attr-defined]
             kernel.last_activity = utcnow()  # type:ignore[attr-defined]
             self.log.info("Kernel started: %s", kernel_id)
-            self.log.debug("Kernel args: %r", kwargs)
+            self.log.debug("Kernel args: %r", {k: v for k, v in kwargs.items() if k != "env"})
```

Comment:
I think this is fine. Another option would be to log what _keys_ are being specified, but not the values. If you see something like $PYTHONPATH or $LD_LIBRARY_PATH etc. in there, it can be an important fact in debugging.